src/pages/BlockDetails.spec.ts

```diff
@@ -0,0 +1,46 @@
+import { mount, config } from '@vue/test-utils';
+import { describe, it, vi, expect } from 'vitest';
+import BlockDetails from '../pages/BlockDetails.vue';
+import { createRouter, createWebHistory } from 'vue-router';
+import { routes } from '../../src/app/router';
+
+const router = createRouter({
+  history: createWebHistory(),
+  routes,
+});
+
+config.global.mocks = {
+  $t: (msg:string) => msg,
+};
+
+vi.mock('vue-i18n', () => ({
+  useI18n: () => ({
+    t: (msg:string) => msg,
+  }),
+}));
+
+// Test conditional rendering:
+// Verify that elements like the block-details-card and transactions-list-page are only rendered when block has a value.
+describe('BlockDetails.vue', async () => {
+  it('renders the block-details-card and block-transactions-list correctly', async () => {
+    router.push('/blocks/606600');
+    await router.isReady();
+    const wrapper = mount(BlockDetails, {
+      global: {
+        plugins: [router],
+      },
+    });
+    await wrapper.vm.$nextTick();
+    await new Promise(resolve => setTimeout(resolve, 10));
```

Comment:
Why this 10ms waiting is needed?

Reply:
This is the minimum time that is needed to let the component load. While the `nextTick()` method should await and let the component load for the component but `BlockTable.vue` using a custom load time was the alternative solution.

Reply:
Hmm, can you elaborate on which process within the component needs to load some extra time? Why single `nextTick` is not enough?